contracts/staking/GoodFundManager.sol

```diff
@@ -285,17 +286,16 @@ contract GoodFundManager is DAOContract {
     /**
      * @dev Mint to users reward tokens which they earned by staking contract
      * @dev _user user to get rewards
-     * @dev _staking staking contract that has implementation of staking reward system
      */
      function mintReward(
-        address _user,
-        StakingContract _staking 
+        address _user
 
      ) public {
         
-        (uint blockReward,uint startBlock,uint endBlock, bool isBlackListed) = getStakingReward(address(_staking));
-        uint amount = _staking._mint(_user);
-        if(amount > 0 && isBlackListed == false){
+        Reward memory staking = rewardsForStakingContract[address(msg.sender)];
+        uint amount = StakingContract(address(msg.sender))._mint(_user);
```

Comment:
if _mint is called externally it should be prefixed with _
_ is for internal

Reply:
changed this _mint method name to userAccounting since it checks user reward amount and return it also it resets it 0